### ISSUE TITLE
[AArch64] Missed FCCMP opportunity

### ISSUE THREAD

**Issue by SamTebbs33** [llvm/llvm-project#60819]
The code

```cpp
bool foo2 (float a, float b)
{
  return __builtin_isnan (a) && __builtin_isnan (b);
}
```
with LLVM `-O2` produces

```asm
foo2(float, float):                              // @foo2(float, float)
        fcmp    s0, s0
        cset    w8, vs
        fcmp    s1, s1
        cset    w9, vs
        and     w0, w8, w9
        ret
```

but GCC can produce:

```asm
foo2(float, float):
        fcmp    s1, s1
        fccmp   s0, s0, 0, vs
        cset    w0, vs
        ret
```

https://godbolt.org/z/e58hM4Pdq

**Comment by efriedma-quic** [comment]
For comparison, the following is optimized:

```
float foo2 (float a, float b, void f())
{
  return __builtin_isnan (a) && __builtin_isnan (b) ? a : b;
}
```

The relevant code is emitConditionalComparison; not sure why it doesn't trigger for the original testcase.

**Comment by SamTebbs33** [comment]
Thanks efriedma, that's very useful. I have someone working on this at the moment.
